File: frontend/src/api/challenge.js
import request from './index'

// ---------- 原有接口 ----------

/**
 * 获取接龙下句候选
 * @param {string} line 当前诗句
 */
export function getNextLines(line) {
  return request.get('/challenge/chain/next', { params: { line } })
}

/**
 * 校验接龙答案（精确匹配）
 * @param {string} current 当前诗句
 * @param {string} answer 用户答案
 */
export function validateChain(current, answer) {
  return request.post('/challenge/chain/validate', { current, answer })
}

/**
 * 生成答题题目（数据库）
 * @param {number} count 题目数量
 */
export function getQuiz(count = 3) {
  return request.get('/challenge/quiz', { params: { count } })
}

// ---------- AI 增强接口 ----------

/**
 * 飞花令 AI 渐进提示
 * @param {string} line 当前诗句
 * @param {number} level 提示级别 1-3
 */
export function getAIHint(line, level = 1) {
  return request.post('/challenge/chain/ai-hint', { line, level }, { timeout: 60000 })
}

/**
 * 飞花令 AI 语义校验
 * @param {string} current 当前诗句
 * @param {string} answer 用户答案
 */
export function aiValidateChain(current, answer) {
  return request.post('/challenge/chain/ai-validate', { current, answer }, { timeout: 60000 })
}

/**
 * 飞花令 AI 对战回合
 * @param {string} current 当前诗句
 */
export function aiChainTurn(current) {
  return request.post('/challenge/chain/ai-turn', { current }, { timeout: 60000 })
}

/**
 * AI 生成多种题型
 * @param {number} count 题目数量
 */
export function getAIQuiz(count = 5) {
  return request.post('/challenge/quiz/ai-generate', null, { params: { count }, timeout: 60000 })
}

/**
 * AI 解析单题
 * @param {object} question 题目对象
 * @param {number} userAnswer 用户选择的选项索引
 */
export function aiExplainAnswer(question, userAnswer) {
  return request.post('/challenge/quiz/ai-explain', { question, user_answer: userAnswer }, { timeout: 60000 })
}

/**
 * AI 答题总结分析
 * @param {Array} results 答题记录列表
 */
export function aiQuizSummary(results) {
  return request.post('/challenge/quiz/ai-summary', { results }, { timeout: 60000 })
}

File: backend/app/routers/challenge.py
"""挑战路由：/api/challenge/*"""
from fastapi import APIRouter, Query
from app.services import challenge_service
from app.models import (
    ValidateRequest,
    ChainHintRequest, ChainAIValidateRequest, ChainAITurnRequest,
    QuizExplainRequest, QuizSummaryRequest,
)
from app.utils import Result

router = APIRouter(prefix="/api/challenge", tags=["挑战"])


# ---------- 原有端点 ----------

@router.get("/chain/next", summary="查找给定诗句的下句候选")
async def next_line(line: str = Query(..., description="完整诗句", example="床前明月光")):
    data = await challenge_service.find_next_lines(line)
    return Result.success(data)


@router.post("/chain/validate", summary="校验接龙答案是否正确（精确匹配）")
async def validate_chain(body: ValidateRequest):
    ok = await challenge_service.validate_chain(body.current, body.answer)
    return Result.success({"correct": ok})


@router.get("/quiz", summary="生成填空选择题（数据库）")
async def quiz(count: int = Query(3, ge=1, le=10, description="题目数量")):
    data = await challenge_service.generate_quiz(count)
    return Result.success(data)


# ---------- AI 增强端点 ----------

@router.post("/chain/ai-hint", summary="飞花令 AI 渐进提示")
async def chain_ai_hint(body: ChainHintRequest):
    data = await challenge_service.ai_hint(body.line, body.level)
    return Result.success(data)


@router.post("/chain/ai-validate", summary="飞花令 AI 语义校验")
async def chain_ai_validate(body: ChainAIValidateRequest):
    data = await challenge_service.ai_validate_chain(body.current, body.answer)
    return Result.success(data)


@router.post("/chain/ai-turn", summary="飞花令 AI 对战回合")
async def chain_ai_turn(body: ChainAITurnRequest):
    data = await challenge_service.ai_chain_turn(body.current)
    return Result.success(data)


@router.post("/quiz/ai-generate", summary="AI 生成多种题型")
async def quiz_ai_generate(count: int = Query(5, ge=1, le=10, description="题目数量")):
    data = await challenge_service.ai_generate_quiz(count)
    return Result.success(data)


@router.post("/quiz/ai-explain", summary="AI 解析单题")
async def quiz_ai_explain(body: QuizExplainRequest):
    data = await challenge_service.ai_explain_answer(body.question, body.user_answer)
    return Result.success(data)


@router.post("/quiz/ai-summary", summary="AI 答题总结分析")
async def quiz_ai_summary(body: QuizSummaryRequest):
    data = await challenge_service.ai_quiz_summary(body.results)
    return Result.success(data)
